packages/inter-protocol/src/psm/psm.js

```diff
@@ -134,6 +135,41 @@ export const start = async (zcf, privateArgs, baggage) => {
   };
   updateMetrics();
 
+  /** @param {Brand<'nat'>} b */
+  const amtShape = b => harden({ brand: b, value: M.nat() });
+
+  /**
+   * @param {ZCFSeat} seat
+   * @param {Omit<MetricsNotification, 'anchorPoolBalance'>} target
+   */
+  const restoreMetricsHook = (seat, target) => {
+    assert(
+      AmountMath.isEmpty(anchorPool.getAmountAllocated('Anchor', anchorBrand)),
+      'cannot restoreMetrics: anchorPool is not empty',
+    );
+    assert(
+      AmountMath.isEmpty(feePool.getAmountAllocated('Minted', stableBrand)),
+      'cannot restoreMetrics: feePool is not empty',
+    );
+    mustMatch(
+      target,
+      harden({
+        feePoolBalance: amtShape(stableBrand),
+        mintedPoolBalance: amtShape(stableBrand),
+        totalAnchorProvided: amtShape(anchorBrand),
+        totalMintedProvided: amtShape(stableBrand),
+      }),
+    );
+    const {
+      give: { Anchor },
+    } = seat.getProposal();
+    stableMint.mintGains({ Minted: target.feePoolBalance }, feePool);
+    atomicRearrange(zcf, harden([[seat, anchorPool, { Anchor }, { Anchor }]]));
```

Comment:
I think something like this is clearer:
```suggestion
    atomicTransfer(zcf, seat, anchorPool, { Anchor });
```